pkg/ocp/nodes_debug.go

```diff
@@ -0,0 +1,307 @@
+	podName string,
+	timeout time.Duration,
+) (*corev1.ContainerStateTerminated, *corev1.Pod, string, error) {
+	pollCtx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	ticker := time.NewTicker(2 * time.Second)
+	defer ticker.Stop()
+
+	var (
+		lastPod    *corev1.Pod
+		terminated *corev1.ContainerStateTerminated
+		waitMsg    string
+	)
+
+	for {
+		select {
+		case <-pollCtx.Done():
+			return nil, nil, "", fmt.Errorf("timed out waiting for debug pod %s to complete: %w", podName, pollCtx.Err())
+		default:
+		}
+
+		// Get pod status using AccessControlClientset
+		pods, getErr := k.AccessControlClientset().Pods(namespace)
+		if getErr != nil {
+			return nil, nil, "", fmt.Errorf("failed to get pods interface: %w", getErr)
+		}
+
+		current, err := pods.Get(pollCtx, podName, metav1.GetOptions{})
+		if err != nil {
+			return nil, nil, "", fmt.Errorf("failed to get debug pod status: %w", err)
+		}
+		lastPod = current
+
+		if status := containerStatusByName(current.Status.ContainerStatuses, NodeDebugContainerName); status != nil {
+			if status.State.Waiting != nil {
+				waitMsg = fmt.Sprintf("container waiting: %s", status.State.Waiting.Reason)
+				// Image pull issues should fail fast.
+				if status.State.Waiting.Reason == "ErrImagePull" || status.State.Waiting.Reason == "ImagePullBackOff" {
+					return nil, nil, "", fmt.Errorf("debug container failed to start (%s): %s", status.State.Waiting.Reason, status.State.Waiting.Message)
+				}
+			}
+			if status.State.Terminated != nil {
+				terminated = status.State.Terminated
+				break
+			}
+		}
+
+		if current.Status.Phase == corev1.PodFailed {
+			break
+		}
+
+		select {
+		case <-pollCtx.Done():
+			return nil, nil, "", fmt.Errorf("timed out waiting for debug pod %s to complete: %w", podName, pollCtx.Err())
+		case <-ticker.C:
+		}
+	}
+
+	return terminated, lastPod, waitMsg, nil
+}
+
+// retrieveLogs retrieves the logs from the debug pod.
+func retrieveLogs(ctx context.Context, k OpenshiftClient, namespace, podName string) (string, error) {
+	logCtx, logCancel := context.WithTimeout(ctx, 30*time.Second)
+	defer logCancel()
+	logs, logErr := k.PodsLog(logCtx, namespace, podName, NodeDebugContainerName, false, 0)
+	if logErr != nil {
+		return "", fmt.Errorf("failed to retrieve debug pod logs: %w", logErr)
+	}
+	return strings.TrimSpace(logs), nil
+}
+
+// processResults processes the debug pod completion status and returns the appropriate result.
+func processResults(terminated *corev1.ContainerStateTerminated, lastPod *corev1.Pod, waitMsg, logs string) (string, error) {
+	if terminated != nil {
+		if terminated.ExitCode != 0 {
+			errMsg := fmt.Sprintf("command exited with code %d", terminated.ExitCode)
+			if terminated.Reason != "" {
+				errMsg = fmt.Sprintf("%s (%s)", errMsg, terminated.Reason)
+			}
+			if terminated.Message != "" {
+				errMsg = fmt.Sprintf("%s: %s", errMsg, terminated.Message)
+			}
+			return logs, errors.New(errMsg)
+		}
+		return logs, nil
```

Comment:
In here can you include the logs in the error message? If you look at how the eventual result to the client is created, the logs will not be returned in the event where there is an error. This makes it hard to know what went wrong

https://github.com/containers/kubernetes-mcp-server/blob/c3bc991237ab9f4aba7f642616f43cd970d27a53/pkg/mcp/mcp.go#L205-L225